isis/src/base/objs/History/History.cpp

```diff
@@ -56,24 +59,19 @@ namespace Isis {
     p_history.addObject(obj);
   }
 
-  /**
-   *
-   */
-  void History::WriteInit() {
+  Blob *History::toBlob(const QString &name) {
     ostringstream ostr;
-    if (p_nbytes > 0) ostr << std::endl;
     ostr << p_history;
     string histStr = ostr.str();
-    int bytes = histStr.size();
+    int nbytes = histStr.size();
 
-    char *temp = p_buffer;
-    p_buffer = new char[p_nbytes+bytes];
-    if (temp != NULL) memcpy(p_buffer, temp, p_nbytes);
+    char *buffer = new char[nbytes];
     const char *ptr = histStr.c_str();
-    memcpy(&p_buffer[p_nbytes], (void *)ptr, bytes);
-    p_nbytes += bytes;
+    memcpy(&buffer[0], (void *)ptr, nbytes);
```

Comment:
Is there a reason to define ptr here? Can this line just be:

```
memcpy(buffer, histStr.c_str(), nbytes);
```

Reply:
Also, I'm fairly certain that `&buffer[0]` is just the same as `buffer` because it's already a pointer.

Reply:
Again just a copy of previous work. I will update this with the suggested changes